### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by abruzgin**

### CODE REVIEW

File: expressServer/app.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 import queryParser from './middlewares/queryParser';
 import router from './routes';
 
 export default function(app) {
   app.use(cookieParser());
   app.use(queryParser());
+  app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({ extended: true }));
+  app.use(session({ 
+    secret: "some_secret",
```
Comment (DraXu3) [inline expressServer/app.js]: Use some practical hash here (and import it from a separate config)

File: expressServer/config/strategies.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+const { consumerKey, consumerSecret, callbackURL: callbackURLTwitter} = config.strategies.twitter;
+const { clientID, clientSecret, callbackURL: callbackURLFb} = config.strategies.facebook;
+const { clientID: googleID, clientSecret: googleSecret, callbackURL: callbackURLGoogle} = config.strategies.google;
+
+export default function(passport) {
+  passport.serializeUser((user, done) => done(null, user));
+
+  passport.deserializeUser((user, done) => done(null, user));
+
+  passport.use('local', new LocalStrategy({
```
Comment (DraXu3) [inline expressServer/config/strategies.js]: Split this file into 3 separate per each strategy (this file should only play a role of facade)

File: expressServer/routes/auth.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import express from "express";
+import passport from "passport";
+import * as auth from '../controllers/auth';
+
+const router = express.Router();
+
+router.post("/", auth.auth);
+router.post("/local", auth.authPassport('local'));
+
+router.get("/twitter", passport.authenticate('twitter'));
```
Comment (DraXu3) [inline expressServer/routes/auth.js:10]: Move to a separate routes files

**Review by DraXu3 — COMMENTED** [review]
Please see some comment to improve

File: expressServer/routes/auth.js (hunk 3 above)
Comment (abruzgin) [reply expressServer/routes/auth.js:10]: @DraXu3 IMO it's logical to put all auth routes, both social-network and local to one file. There's no need to separate them

Comment (DraXu3) [reply expressServer/routes/auth.js:10]: @AlexBryuzgin It's not such important point to be resolved in the hometask's scope, so let's skip it. But I pay attention to the question: what type of grouping is more solid: per routes or per socials (and then re-think it with a case when we want to rename some social in future)?

**Pull request merged**